core/shared/src/main/scala/scalaz/zio/TaskR.scala

```diff
@@ -0,0 +1,370 @@
+
+  /** See [[scalaz.zio.ZIO.mergeAll]]
+   */
+  final def mergeAll[R, A, B](in: Iterable[TaskR[R, A]])(zero: B)(f: (B, A) => B): TaskR[R, B] =
+    ZIO.mergeAll(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.mergeAllPar]]
+   */
+  final def mergeAllPar[R, A, B](in: Iterable[TaskR[R, A]])(zero: B)(f: (B, A) => B): TaskR[R, B] =
+    ZIO.mergeAllPar(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.provide]]
+   */
+  final def provide[R, A](r: R): TaskR[R, A] => Task[A] =
+    ZIO.provide(r)
+
+  /** See [[scalaz.zio.ZIO.raceAll]]
+   */
+  final def raceAll[R, R1 <: R, A](zio: TaskR[R, A], ios: Iterable[TaskR[R1, A]]): TaskR[R1, A] =
+    ZIO.raceAll(zio, ios)
+
+  /** See [[scalaz.zio.ZIO.reduceAll]]
+   */
+  final def reduceAll[R, R1 <: R, A](a: TaskR[R, A], as: Iterable[TaskR[R1, A]])(f: (A, A) => A): TaskR[R1, A] =
+    ZIO.reduceAll(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.reduceAllPar]]
+   */
+  final def reduceAllPar[R, R1 <: R, A](a: TaskR[R, A], as: Iterable[TaskR[R1, A]])(f: (A, A) => A): TaskR[R1, A] =
+    ZIO.reduceAllPar(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.require]]
+   */
+  final def require[A](error: Throwable): Task[Option[A]] => Task[A] =
+    ZIO.require(error)
+
+  /** See [[scalaz.zio.ZIO.reserve]]
+   */
+  def reserve[R, A, B](reservation: TaskR[R, Reservation[R, Throwable, A]])(use: A => TaskR[R, B]): TaskR[R, B] =
+    ZIO.reserve(reservation)(use)
+
+  /** See [[scalaz.zio.ZIO.runtime]]
+   */
+  final def runtime[R]: ZIO[R, Nothing, Runtime[R]] = ZIO.runtime
+
+  /** See [[scalaz.zio.ZIO.succeed]]
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** See [[scalaz.zio.ZIO.succeedLazy]]
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** See [[scalaz.zio.ZIO.supervise]]
+   */
+  final def supervise[R, A](zio: TaskR[R, A]): TaskR[R, A] =
+    ZIO.supervise(zio)
+
+  /** See [[scalaz.zio.ZIO.supervised]]
+   */
+  final def supervised[R, A](zio: TaskR[R, A]): TaskR[R, A] =
+    ZIO.supervised(zio)
+
+  /** See [[scalaz.zio.ZIO.superviseWith]]
+   */
+  final def superviseWith[R, A](
+    zio: TaskR[R, A]
+  )(supervisor: IndexedSeq[Fiber[_, _]] => ZIO[R, Nothing, _]): TaskR[R, A] =
+    ZIO.superviseWith(zio)(supervisor)
+
+  /** See [[scalaz.zio.ZIO.suspend]]
+   */
+  final def suspend[R, A](io: => TaskR[R, A]): TaskR[R, A] =
+    ZIO.suspend(io)
+
+  /** See [[scalaz.zio.ZIO.swap]]
+   */
+  final def swap[R, A, B](implicit ev: R <:< (A, B)): TaskR[R, (B, A)] =
+    ZIO.swap
+
+  /** See [[scalaz.zio.ZIO.uninterruptible]]
+   */
+  final def uninterruptible[R, A](zio: TaskR[R, A]): TaskR[R, A] =
+    ZIO.uninterruptible(zio)
+
+  /** See [[scalaz.zio.ZIO.uninterruptibleMask]]
+   */
+  final def uninterruptibleMask[R, A](k: ZIO.InterruptStatusRestore => TaskR[R, A]): TaskR[R, A] =
+    ZIO.uninterruptibleMask(k)
+
+  /** See [[scalaz.zio.ZIO.unsandbox]]
+   */
+  final def unsandbox[R, A](v: ZIO[R, Cause[Throwable], A]): TaskR[R, A] = ZIO.unsandbox(v)
```

Comment:
I would delete this method from `TaskR`, because the input value `v` cannot be typed as a `TaskR`, since `Cause[Throwable]` is not a subtype of `Throwable`. If a user wants to use this function, they have to go to `IO` or `ZIO`.